lib/plugins/physics.js

```diff
@@ -220,6 +222,47 @@ function inject (bot) {
     return vel.x * vel.x + vel.z * vel.z
   }
 
+  function liquidPush (pos) {
+    const flow = liquidFlow(pos);
+    const push = flow.clone().scaled(physics.liquidPush);
+    push.y = push.y/physics.gravity;
+    return push
+  }
+
+  function liquidFlow (pos) {
+    const flow = new Vec3(0,0,0);
+    const block = bot.blockAt(pos)
+    const base = block.metadata;
+    const id = block.type;
+    if(base > 7) {
+      base = 0;
+      flow.set(0,-1,0);
+    }
+    const depth = level => {
+      if(level > 7) return 0;
+      return level; 
+    }
+    const Blocks = {
+      'x+': bot.blockAt(pos.offset(1,0,0)),
+      'x-': bot.blockAt(pos.offset(-1,0,0)),
+      'z+': bot.blockAt(pos.offset(0,0,1)),
+      'z-': bot.blockAt(pos.offset(0,0,-1))
+    }
+    const waterBlocks = {}
+    for(block of Object.entries(Blocks)){
+      if(Blocks[key].type == id) waterBlocks[key] = depth(Blocks[key].metadata)
+    }
+    if(waterBlocks['x+'] !== null && waterBlocks['x+'] > base) flow = flow.add(1,0,0);
+    if(waterBlocks['x-'] !== null && waterBlocks['x-'] > base) flow = flow.add(-1,0,0);
+    if(waterBlocks['z+'] !== null && waterBlocks['z+'] > base) flow = flow.add(0,0,1);
+    if(waterBlocks['z-'] !== null && waterBlocks['z-'] > base) flow = flow.add(0,0,-1);
+    if(waterBlocks['x+'] !== null && waterBlocks['x+'] < base && flow.x == 0) flow = flow.add(-1,0,0)
+    if(waterBlocks['x-'] !== null && waterBlocks['x-'] < base && flow.x == 0) flow = flow.add(1,0,0);
+    if(waterBlocks['z+'] !== null && waterBlocks['z+'] < base && flow.z == 0) flow = flow.add(0,0,-1)
+    if(waterBlocks['z-'] !== null && waterBlocks['z-'] < base && flow.z == 0) flow = flow.add(0,0,1);
```

Comment:
flow is a constant, add does inplace editing of the vector, removing the `flow = ` in all of these lines
also, it looks like you didnt run standard --fix after doing wvvfle's requested changes

Reply:
oops

Reply:
I also need to change something